pwiz_tools/Skyline/Util/Xml.cs

```diff
@@ -75,7 +75,7 @@ public XmlSchema GetSchema()
             return null;
         }
 
-        public void ReadXml(XmlReader reader)
+        public virtual void ReadXml(XmlReader reader)

```

Comment:
There is already a virtual method ValidateLoad() in which you could do this. I also think we should probably do it for everything in GetDefaults() so that we are guaranteed that all of the defaults in the list that are Equals a value returned from GetDefaults() are also ReferenceEquals().